orttraining/orttraining/python/training/_ortmodule_training_manager.py

```diff
@@ -122,6 +122,7 @@ def backward(ctx, *grad_outputs):
 
                 backward_outputs = C.OrtValueVector()
                 self._execution_agent.run_backward(backward_inputs, backward_outputs, ctx.run_info.state)
+                del ctx.run_info.state
```

Comment:
I believe we should work out the lifecycle of little bit better.
 
It is not expected to have `_ORTModuleFunction` controlling the lifecycle of `PartialGraphExecutionState` inside `RunStateInfo`. It is not _ORTModuleFunction responsibility to do this house keeping. Code coupling grows exponentially and and that adds up to the hard-to-find crashes

Reply:
Max batch size on NV-BERT goes down from 34 to less than 30, if NOT for that line. PartialGraphExecutionState is created in ORTModule, its lifecycle needs to also be managed on ORTModule side.

Reply:
Ok, great, thanks!

Reply:
The comment helps.
I will create a work item to see how we can improve this in future tasks